### PR TITLE
Version tracker

### PR DESCRIPTION
**Pull request by warlof**
Instead having separate place to be able to review package version, this will provide all information into the same location, for both core and plugin packages.

This will depends on a Provider update which have to be inherited by all SeAT package provider in order to be able to provide proper information and implement auto-discovery.

This SeAT provider should be able to return enough information to be able to build some kind of marketplace and make plugin installation easier.

The rendered list is making live checking together with packagist API in order to figure if or not an installed package is outdated. This is figured by displaying a green tick (up-to-date), red cross (outdated) or yellow question mark (pending check).

image
image

Rendered table from copy button :
``````
```
| Vendor  | Package Name   | Installed Version |
| ------- | -------------- | ----------------- |
| eveseat | api            | 3.0.5             |
| eveseat | console        | 3.0.2             |
| eveseat | eveapi         | 3.0.9             |
| eveseat | notifications  | 3.0.2             |
| eveseat | services       | 3.0.4             |
| eveseat | web            | 3.0.12            |
| warlof  | seat-teamspeak | 3.0.9             |
```
``````

| Vendor  | Package Name   | Installed Version |
| ------- | -------------- | ----------------- |
| eveseat | api            | 3.0.5             |
| eveseat | console        | 3.0.2             |
| eveseat | eveapi         | 3.0.9             |
| eveseat | notifications  | 3.0.2             |
| eveseat | services       | 3.0.4             |
| eveseat | web            | 3.0.12            |
| warlof  | seat-teamspeak | 3.0.9             |

**Depends**: eveseat/services#80

### CODE REVIEW

**Comment by warlof** [comment]
Ok, so I just implemented a way to retrieve changelog for a package using either API or a Markdown file.
This is made possible using this optional stack : https://github.com/eveseat/services/blob/a16ac028e411d4dd933d47f7bb8557c1ec199f23/src/AbstractSeatPlugin.php#L70-L117

The result is either one or the other screen bellow. In case none of those method are overloaded, the button is not available.

image

image

(Saying that, we really have crap release styling on Github for core :p)

File: src/Http/Controllers/Configuration/SeatController.php [hunk 1]
```diff
@@ -107,4 +113,219 @@ public function getApprovedSDE()
...
+
+                // retrieve releases from provided API endpoint
+                $client = new Client();
+                $response = $client->request('GET', $uri);
+
+                // decode the response
+                $json_object = json_decode($response->getBody());
+
+                // spawn a new Markdown parser
+                $parser = new Parsedown();
```
Comment (leonjza) [inline src/Http/Controllers/Configuration/SeatController.php]: Please add `$parser->setSafeMode(true);`

File: src/Http/Controllers/Configuration/SeatController.php [hunk 2]
```diff
@@ -107,4 +113,219 @@ public function getApprovedSDE()
...
+    private function getChangelogFromFile(string $uri)
+    {
+        try {
+            return cache()->remember($this->getChangelogCacheKey($uri), 30, function () use ($uri) {
+                // retrieve changelog from provided uri
+                $client = new Client();
+                $response = $client->request('GET', $uri);
+
+                // spawn a new Markdown parser
+                $parser = new Parsedown();
```
Comment (leonjza) [inline src/Http/Controllers/Configuration/SeatController.php]: Please add $parser->setSafeMode(true);

File: src/WebServiceProvider.php [hunk 3]
```diff
@@ -412,4 +411,125 @@ private function configure_api()
...
+    }
+
+    /**
+     * Return the plugin author e-mail address.
+     *
+     * @return string|null
+     */
+    public function getAuthorMailAddress(): ?string
+    {
+        return 'theninjabag@gmail.com';
```
Comment (leonjza) [inline src/WebServiceProvider.php]: I dont think this information (getAuthorEveCharacterID(), getAuthorName() and getAuthorMailAddress(), getAuthorSlackNickname()) is useful. A github link is enough.

File: src/resources/views/configuration/settings/partials/packages/changelog/body.blade.php [hunk 4]
```diff
@@ -0,0 +1 @@
+<div>{!! $body !!}</div>
```
Comment (leonjza) [inline src/resources/views/configuration/settings/partials/packages/changelog/body.blade.php]: Missing newline please.

**Comment by leonjza** [comment]
Nice work :D

**Comment by herpaderpaldent** [comment]
i love the clipboard functionality <3 that is going to help in support cases :D

**Comment by leonjza** [comment]
LGTM

**Pull request merged**